syntax = "proto3";

message Vector3f {
  float x = 1;
  float y = 2;
  float z = 3;
}

message Color {
  float r = 1;
  float g = 2;
  float b = 3;
}

message Colors {
  repeated Color colors = 1;
}

message PointCloud {
  repeated Vector3f points = 1;
}

message Line {
  Vector3f start = 1;
  Vector3f end = 2;
}

message Lines {
  repeated Line lines = 1;
}

message Linestrip {
  repeated Vector3f points = 1;
}

message Plane {
  Vector3f center = 1;
  Vector3f normal = 2;
}

message Cuboid {
  message MinMax {
    Vector3f min = 1;
    Vector3f max = 2;
  }

  message CenterRadius {
    Vector3f center = 1;
    Vector3f radius = 2;
  }

  oneof bounds {
    MinMax min_max = 1;
    CenterRadius center_radius = 2;
  }
}

message Sphere {
  Vector3f center = 1;
  float radius = 2;
}
